src/types/uint.rs

```diff
@@ -93,6 +115,16 @@ macro_rules! impl_uint {
       }
     }
 
+    impl<'a> From<&'a [u8]> for $name {
```

Comment:
Hm, I guess since `AsRef<[u8]>` _could_ be added to `u64` eventually :(
We'll have to wait for specialization then.